### PR TITLE
 MC-18096: add entity to checker, ensures result is set on task

### PR DESCRIPTION
**Pull request by patrickspensieri**
#### Issue
- Several operations like _Reset password_ only return required information via sensitive notifications
- The result is not set on the task, meaning an API user will never know what the new password is for their instance
- (In most cases, sensitive information like passwords and ssh keys is not returned by azure when fetching the entity)

#### Solution
**1. Short term** (for operations returning sensitive credentials)
- Add an `entity` to the `Checker`, which core uses to set the result on a task
- Update exclude annotations, add `scope = Exclude.Scope.AUDIT` to allow the sensitive information to be serialized on task results, see tests for more detail

**2. Long term**
Opened [MC-18704](https://cloudops.atlassian.net/browse/MC-18704). All operations should return a task result. Assume an API user creates a network, they may want to know the resulting network ID.

#### Test cases
- Unit tested
- Manually tested

#### UI changes
No changes

#### Related PRs
- https://github.com/cloudops/cloudmc-azure-plugin/pull/263

### CODE REVIEW

File: source/includes/azure/_clusters.md [hunk 1]
```diff
@@ -129,6 +129,34 @@ curl -X POST \
...
+> The above command(s) return(s) JSON structured like this:
+
+```js
+{
+	"data": {
+		"id": "ba0d9e44-ce89-4cc0-9c90-da1dcde1a8ac",
+		"status": "SUCCESS",
+		"created": "2022-06-23T13:11:07.03164-04:00",
+		"result": {
+			"sshKey": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCZcbdHKq1YK4vV5gtemy5SzngDhU3mSGhsxyBe4uKvEwk4shzRBPRc8uvOm0KrzB9KkbuxR3OJ1TbFVijnmvrOGIacN7BsJENBcBsrjjHW3vNPF0ZaJWts+lUPL+6BcuDTNUSyDTpAI1Xs/a3P5E7kOQumiwjlXIwuyJyGIk5Lt2FmZgDSCuZ4fjuUGgQq27RDya7G/eB5eKD9ohmXM567VHWvGQKUV4RdNDM0W4YuQYCAsaLwaKhFMOxr1AK5YOgyJYVVZn280bOx2qDyp6fTN49UTbkOtcQz+S8d2PRLaplY3QVPtelV47s2Gawgl8i3jiXGBbMV6WllLM+ujZy7 ",
```
Comment (fyrnaga) [inline source/includes/azure/_clusters.md:155]: Should have the username as well?

File: source/includes/azure/_clusters.md [hunk 2]
```diff
@@ -150,6 +178,14 @@ Optional | &nbsp;
 ------- | -----------
 `rootUsername`<br/>*string* | The user name to create a root user on cluster.
 
+Attributes | &nbsp;
+------- | -----------
+`id`<br/>*string* | The task's ID.
+`status`<br/>*string* | The task's status.
+`created`<br/>*Date* | The task's creation date.
+`result`<br/>*Object* | The task result, containing details of the newly created cluster.
+`result.sshKey`<br/>*string* | The SSH key for the newly created cluseter.
```
Comment (fyrnaga) [inline source/includes/azure/_clusters.md]: same here

**Review by evanluc — APPROVED** [review]

File: source/includes/azure/_clusters.md (hunk 2 above)
Comment (patrickspensieri) [reply source/includes/azure/_clusters.md]: Fixed [here](https://github.com/cloudops/cloudmc-api-docs/pull/558/commits/15b09b5a037e690c60b21d17b3efac632561c80c)

File: source/includes/azure/_clusters.md (hunk 1 above)
Comment (patrickspensieri) [reply source/includes/azure/_clusters.md:155]: You are correct, fixed [here](https://github.com/cloudops/cloudmc-api-docs/pull/558/commits/15b09b5a037e690c60b21d17b3efac632561c80c)

**Review by fyrnaga — APPROVED** [review]

**Pull request merged**